### PR TITLE
doc: Bluetooth RPC Host related documentation updates

### PR DESCRIPTION
**Pull request by ppryga-nordic**
There were updated documentation related to Bluetooth RPC host interface in following parts of SDK:
- nordic-bt-rpc snippet
- BLE RPC subsystem
- ipc_radio
- peripheral_uart

### CODE REVIEW

**Review by dchat-nordic — APPROVED** [review]

File: snippets/nordic-bt-rpc/README.rst [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
+#########################################################################
 
 Overview
 ********
 
-This snippet allows users to build Bluetooth applications with Bluetooth stack running on Radio
-core. The communication between an application and Bluetooth stack is handled over the Bluetooth
-RPC subsystem.
+This snippet allows you to build Bluetooth applications with Bluetooth stack running on a core dedicated for radio communication.
+For example in the nRF54H20 SoC it is radio core.
```
Comment (peknis) [inline snippets/nordic-bt-rpc/README.rst:10]:
```suggestion
For example in the nRF54H20 SoC, it is the radio core.
```

File: snippets/nordic-bt-rpc/README.rst [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
+Supported SoCs
+**************
+
+Currently, the following SoCs from Nordic Semiconductor are supported for use with the snippet:
+
+* :ref:`zephyr:nrf5340dk_nrf5340`
+* :ref:`zephyr:nrf54h20dk_nrf54h20`
+
+.. note::
+   For the nRF54H20 SoC the snippet modifies the memory map to make room for settings storage for the radio core.
```
Comment (peknis) [inline snippets/nordic-bt-rpc/README.rst:22]:
```suggestion
   On the nRF54H20 SoC, the snippet modifies the memory map to make room for settings storage for the radio core.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: snippets/nordic-bt-rpc/README.rst (hunk 1 above)
Comment (ppryga-nordic) [reply snippets/nordic-bt-rpc/README.rst]: Corrected

File: snippets/nordic-bt-rpc/README.rst (hunk 2 above)
Comment (ppryga-nordic) [reply snippets/nordic-bt-rpc/README.rst]: Corrected

**Review by peknis — APPROVED** [review]

**Pull request merged**